packages/plugin-commands-outdated/src/outdated.ts

```diff
@@ -289,26 +326,32 @@ export function renderCurrent ({ current, wanted }: OutdatedPackage) {
   return `${output} (wanted ${wanted})`
 }
 
-export function renderLatest (outdatedPkg: OutdatedWithVersionDiff): string {
+export function renderLatest (outdatedPkg: OutdatedWithVersionDiff, withoutChalk = false): string {
   const { latestManifest, change, diff } = outdatedPkg
   if (latestManifest == null) return ''
   if (change === null || (diff == null)) {
     return latestManifest.deprecated
-      ? chalk.redBright.bold('Deprecated')
+      ? withoutChalk ? 'Deprecated' : chalk.redBright.bold('Deprecated')
       : latestManifest.version
   }
 
+  if (withoutChalk) {
+    return diff.flat().join('.')
+  }
+
   return colorizeSemverDiff({ change, diff })
 }
 
-export function renderDetails ({ latestManifest }: OutdatedPackage) {
+export function renderDetails ({ latestManifest }: OutdatedPackage, withoutChalk = false) {
```

Comment:
create a new function for json

Reply:
same here https://github.com/pnpm/pnpm/pull/5582#discussion_r1014605452